### DISCUSSION TITLE
Object examples do not match the type definition

### DISCUSSION THREAD

**Discussion by Amorim33** [scalar/scalar#1051] [Q&A]
## Description

We are using Hono for the API, and [Hono Zod Openapi](https://hono.dev/snippets/zod-openapi) to generate the OpenAPI specs.

When creating a field that is a `Record<string, unknown>`, Scalar generates an example with the shape: `field: {}`.
However, when creating a field that is a `Record<string, string>`, Scalar generates an example with the shape: `field: ""`

![image](https://github.com/scalar/scalar/assets/42624869/08b3c220-4e18-4883-aa4d-eb0299e04445)

In the second case, the **example is not valid** since the field is not a string.

## Expected behavior
It would be useful to have examples of keys and values when an object is defined. Also, maybe an empty object as an example could be confusing, since in most of the cases an empty object would not be a valid input and the field could be nullable.

### Redoc (example)
![image](https://github.com/scalar/scalar/assets/42624869/15236ccb-ce25-45a1-a7d2-9d515165ddfc)

- Does it make sense?

## Steps to reproduce

- Add `@scalar/hono-api-reference` to the project;
- Add OpenAPI fields to the OpenAPI json schema.

### Package version
```json
{
    "@scalar/hono-api-reference": "^0.3.56",
}
```
### OpenAPI fields
```json
{
  "additionalAttributes": {
        "type": "object",
        "nullable": true,
        "additionalProperties": {
            "nullable": true
        },
        "description": "Atributos adicionais do item segurado. **O campo será removido futuramente, utilize o campo metadata**.",
        "deprecated": true
    },
    "metadata": {
        "type": "object",
        "nullable": true,
        "additionalProperties": {
            "type": "string",
            "minLength": 1,
            "maxLength": 500
        },
        "description": "Metadados do item segurado: você pode especificar até 50 chaves, com nomes de até 40 caracteres e valores de até 500 caracteres.",
        "maxItems": 50
    },
}
```

**Comment by Amorim33** [comment]
If this aligns with the library's goals, I'm fully committed to resolving this issue. 🫡

**Reply by hanspagel** [reply]
Thanks for the report! Yes, let’s fix this. If you’d like to send a PR, I’d be happy to review it! 🤝 Don’t hesitate to ask for help here or on Discord. ✌️

**Reply by Amorim33** [reply]
Very nice!
Thanks for the opportunity, I'll work on it this week!😁

**Comment by Amorim33** [comment]
Done https://github.com/scalar/scalar/pull/1054
